runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -44,111 +42,12 @@
+static struct shared_heap {
   void* base;
   size_t size;
   size_t cur_offset;
   pthread_mutex_t alloc_lock;
-} shared_heap_t;
-
-static shared_heap_t* heaps;
-
-static int get_num_heaps(void) {
-  static int num_heaps = -1;
-
-  if (num_heaps < 0) {
-    //
-    // Initialize: without multiple NUMA domains we definitely don't
-    // need multiple heaps.  Even with multiple NUMA domains we might
-    // not need multiple heaps, if other constraints aren't satisfied.
-    //
-    void* heap_base;
-    size_t heap_size;
-
-    chpl_comm_get_registered_heap(&heap_base, &heap_size);
-    if (heap_base == NULL) {
-      num_heaps = 0;
-    } else {
-      num_heaps = chpl_topo_getNumNumaDomains();
-      if (num_heaps < 1) {
-        num_heaps = 1;
-      } else if (!chpl_mem_impl_localizes()) {
-        num_heaps = 1;
-      }
-    }
-  }
-
-  return num_heaps;
-}
-
-static inline int get_nearby_heap(void) {
-  if (get_num_heaps() == 1) {
-    return 0;
-  } else {
-    int hpi = chpl_topo_getThreadLocality();
-    if (!isActualSublocID(hpi) || hpi >= get_num_heaps()) {
-      chpl_internal_error("unexpected heap index");
-    }
-    return hpi;
-  }
-}
-
-
-// helper routine to initialize the table of heaps
-static void setupLocalizedHeaps(void* heap_base, size_t heap_size) {
-  const int num_heaps = get_num_heaps();
-  int hpi;
-  char* subchunk_base;
-  size_t* subchunk_sizes;
-
-  heaps = (shared_heap_t*) CHPL_JE_MALLOC(num_heaps * sizeof(*heaps));
-  if (heaps == NULL) {
-    chpl_internal_error("cannot allocate heaps");
-  }
-
-  if (num_heaps <= 1) {
-    heaps[0].base = heap_base;
-    heaps[0].size = heap_size;
-    heaps[0].cur_offset = 0;
-    if (pthread_mutex_init(&heaps[0].alloc_lock, NULL) != 0) {
-      chpl_internal_error("cannot init chunk_alloc lock");
-    }
-
-    return;
-  }
-
-  subchunk_sizes =
-    (size_t*) CHPL_JE_MALLOC(num_heaps * sizeof(*subchunk_sizes));
-  if (subchunk_sizes == NULL) {
-    chpl_internal_error("cannot allocate subchunk_sizes");
-  }
-
-  chpl_topo_setMemSubchunkLocality(heap_base, heap_size, true,
-                                   subchunk_sizes);
-
-  subchunk_base = (char*) heap_base;
-  for (hpi = 0; hpi < num_heaps; hpi++) {
-    heaps[hpi].base = subchunk_base;
-    heaps[hpi].size = subchunk_sizes[hpi];
-    heaps[hpi].cur_offset = 0;
-    if (pthread_mutex_init(&heaps[hpi].alloc_lock, NULL) != 0) {
-      chpl_internal_error("cannot init chunk_alloc lock");
-    }
-    subchunk_base += heaps[hpi].size;
-
-    //
-    // The chpl_topo_setMemSubchunkLocality() call above isn't
-    // having the effect we desire, for reasons that are mostly but
-    // not completely understood.  So for now, to get the locality
-    // we want, touch the pages into existence while executing on
-    // the desired NUMA domains.
-    //
-    chpl_topo_touchMemFromSubloc(heaps[hpi].base, heaps[hpi].size, true, hpi);
-  }
-}
+} heap; // static, will be "zero" initialized automatically
```

Comment:
Do we really need to tell programmers that file=level C variables are zero-initialized?

Reply:
Note that this is a reversion back to what was there. 

That said it could probably be removed. I originally added it to note why there was no explicit initializer: https://github.com/chapel-lang/chapel/pull/3388#r54660489

Reply:
It does seem odd to have this comment on this one variable and not have a similar one on all the rest.  I won't die if it stays, though.

Reply:
Will remove it